### ISSUE TITLE
Feature request :  Improvement providers t411

### ISSUE THREAD

**Issue by devildantsky** [SickRage/SickRage#2835]
Hello,

Currently the implementation of the provider t411 is not complete.
For example:

The animated series twin star exorcists contains 37 episodes (present on t411), but sickrage only a few episodes.
The reason why episodes are not found is that the implementation is not complete
T411 has a complex API with category and term.

category list for video : 
{
   "210":{
      **"id":"210",**
      "pid":"0",
      **"name":"Film\/Vid\u00e9o",**
      "cats":{
         "402":{
            "id":"402",
            "pid":"210",
            "name":"Vid\u00e9o-clips"
         },
         "433":{
            "id":"433",
            "pid":"210",
            "name":"S\u00e9rie TV"
         },
         "455":{
            "id":"455",
            "pid":"210",
            "name":"Animation"
         },
         "631":{
            "id":"631",
            "pid":"210",
            "name":"Film"
         },
         "633":{
            "id":"633",
            "pid":"210",
            "name":"Concert"
         },
         "634":{
            "id":"634",
            "pid":"210",
            "name":"Documentaire"
         },
         "635":{
            "id":"635",
            "pid":"210",
            "name":"Spectacle"
         },
         "636":{
            "id":"636",
            "pid":"210",
            "name":"Sport"
         },
         "637":{
            **"id":"637",**
            "pid":"210",
            **"name":"Animation S\u00e9rie"**
         },
         "639":{
            "id":"639",
            "pid":"210",
            "name":"Emission TV"
         }
      }
   }

term for search specific Season and episode : 
"45":{
         "type":"S\u00e9rieTV - Saison",
         "mode":"single",
         "terms":{
            "982":"Saison 15",
            "993":"Saison 27",
            "972":"Saison 05",
            "983":"Saison 16",
            "994":"Saison 25",
            "973":"Saison 06",
            "984":"Saison 17",
            "995":"Saison 28",
            "974":"Saison 07",
            "985":"Saison 18",
            "996":"Saison 29",
            "975":"Saison 08",
            "986":"Saison 19",
            "997":"Saison 30",
            "976":"Saison 09",
            "987":"Saison 20",
            "998":" S\u00e9rie int\u00e9grale",
            "977":"Saison 10",
            "988":"Saison 21",
            "1068":"Hors saison",
            "978":"Saison 11",
            "989":"Saison 22",
            "1144":"Non communiqu\u00e9",
            "968":"Saison 01",
            "979":"Saison 12",
            "990":"Saison 23",
            "969":"Saison 02",
            "980":"Saison 13",
            "991":"Saison 24",
            "970":"Saison 03",
            "981":"Saison 14",
            "992":"Saison 26",
            "971":"Saison 04"
         }
      },
      "46":{
         "type":"S\u00e9rieTV - Episode",
         "mode":"single",
         "terms":{
            "1100":"Episode 43",
            "948":"Episode 11",
            "959":"Episode 22",
            "1111":"Episode 54",
            "1090":"Episode 33",
            "937":"Episode 01",
            "1101":"Episode 44",
            "949":"Episode 12",
            "960":"Episode 23",
            "1112":"Episode 55",
            "1091":"Episode 34",
            "938":"Episode 02",
            "1102":"Episode 45",
            "950":"Episode 13",
            "961":"Episode 24",
            "1113":"Episode 56",
            "1092":"Episode 35",
            "939":"Episode 03",
            "1103":"Episode 46",
            "951":"Episode 14",
            "1114":"Episode 57",
            "962":"Episode 25",
            "1093":"Episode 36",
            "940":"Episode 04",
            "952":"Episode 15",
            "1104":"Episode 47",
            "1115":"Episode 58",
            "963":"Episode 26",
            "1094":"Episode 37",
            "941":"Episode 05",
            "953":"Episode 17",
            "1105":"Episode 48",
            "1116":"Episode 59",
            "964":"Episode 27",
            "1095":"Episode 38",
            "942":"Episode 06",
            "954":"Episode 16",
            "1106":"Episode 49",
            "1117":"Episode 60",
            "965":"Episode 28",
            "1096":"Episode 39",
            "943":"Episode 07",
            "955":"Episode 18",
            "1107":"Episode 50",
            "1143":"Non communiqu\u00e9",
            "966":"Episode 29",
            "1097":"Episode 40",
            "944":"Episode 08",
            "956":"Episode 19",
            "1108":"Episode 51",
            "967":"Episode 30",
            "1098":"Episode 41",
            "946":"Episode 09",
            "957":"Episode 20",
            "1109":"Episode 52",
            "1088":"Episode 31",
            "1099":"Episode 42",
            "947":"Episode 10",
            "958":"Episode 21",
            "1110":"Episode 53",
            "1089":"Episode 32",
            "936":" Saison compl\u00e8te"
         }
      }

exemple  with video categogy : 
/torrents/search/twin+star+exorcists?**cat=210**

exemple with subcategory Animation series : 
/torrents/search/twin+star+exorcists?cat=210&**subcat=637**

exemple with season : 
/torrents/search/twin+star+exorcists?cat=210&subcat=637&**term[45][]=968**

exemple with season and episode : 
/torrents/search/twin+star+exorcists?cat=210&subcat=637**&term[45][]=968&term[46][]=937**

basically : 
category use : 210
sub category : 
     -  433 for the normal series
     -  637 for the anime series
term search season : 
  [46][] = termlist (see the partial json term for search specific Season and episode)

term search episode : 
  [45][] = termlist (see the partial json term for search specific Season and episode)

(you can see the api information here : https://api.t411.li)

Do you think it would be possible to do this implementation to improve the search ?
best regards

**Comment by SickRage2** [comment]
Thanks for the issue report! Before a real human comes by, please make sure your report has all the below criteria checked

- [ ] Include basic information: Branch/Commit, OS, What you did, What happened, What you expected
- [ ] Enable debug logging (be sure to disable after the bug is fixed)
- [ ] Post debug logs, either inline (for smaller logs) or using gist

Please make sure you also read [how to create an issue](https://github.com/SickRage/sickrage-issues/blob/master/README.md) and followed all of the steps. 

The title should describe your issue. Having "SR not working" or "I get this bug" for 100 issues, isn't really helpful. We will close issues if there isn't enough information.

Sometimes the devs may seem like grunts and respond with short answers. This isn't (always) because the dev hates you, but because he's on mobile or busy fixing bugs. If something isn't clear, please let us know, and this bot may get updated to automatically answer you.

Thanks!

**Comment by devildantsky** [comment]
Sorry i see today that exist the specific section for the feature request, i close this and reopen inside the good section.

best regards
